nomad/job_endpoint.go

```diff
@@ -246,7 +246,7 @@ func (j *Job) Register(args *structs.JobRegisterRequest, reply *structs.JobRegis
 	//   - reading kv store of each group
 	//   - establishing consul connect services
 	checkConsulToken := func(usages map[string]*structs.ConsulUsage) error {
-		if j.srv.config.ConsulConfig.AllowsUnauthenticated() {
+		if j.srv.config.GetDefaultConsul().AllowsUnauthenticated() {
```

Comment:
We could modify the import to be a `map[string][string]*structs.ConsulUsage` where the first key is the Consul cluster, but I think this is used for the deprecated `-consul-token` right? So probably not worth modifying it.

Reply:
Exactly!